pkg/storage/unified/resource/broadcaster_test.go

```diff
@@ -104,3 +105,42 @@ func TestCache(t *testing.T) {
 	// slice should return all values
 	require.Equal(t, []int{4, 5, 6, 7, 8, 9, 10, 11, 12, 13}, c.Slice())
 }
+
+func TestBroadcaster(t *testing.T) {
+	ctx := testutil.NewDefaultTestContext(t)
+	ch := make(chan int)
+	input := []int{1, 2, 3}
+	go func() {
+		for _, v := range input {
+			ch <- v
+		}
+	}()
+	t.Cleanup(func() {
+		close(ch)
+	})
+
+	b, err := NewBroadcaster(ctx, func(out chan<- int) error {
+		go func() {
+			for v := range ch {
+				out <- v
+			}
+		}()
+		return nil
+	})
+	require.NoError(t, err)
+
+	sub, err := b.Subscribe(ctx)
+	require.NoError(t, err)
+
+	count := 0
+	for v := range sub {
```

Comment:
I don't think this test works correctly. `sub` channel is only closed because used context times out. If it wasn't, test would be stuck here.

Reply:
I agree. Fixing